src/test/compile-fail/E0408.rs

```diff
@@ -12,7 +12,7 @@ fn main() {
     let x = Some(0);
 
     match x {
-        Some(y) | None => {} //~ ERROR E0408
-        _ => ()
+        Some(y) | None => {} //~ ERROR variable `y` from pattern #1 is not bound in pattern #2
+        _ => () //~^ NOTE pattern doesn't bind `y`
```

Comment:
Generally, we use "//~|" and not "//~^" when it's on the same line. I mean that the note is thrown by the same line than the error, so it's the "same" issue. With this way to write it, we have the feeling that it's two different errors.

Also, can you align "//~^ NOTE pattern doesn't bind `y`" with "//~ ERROR variable `y` from pattern #1 is not bound in pattern #2" please?